decryptor/mysql/prepared_statements.go

```diff
@@ -281,3 +285,103 @@ func (m *mysqlBoundValue) Encode() (encoded []byte, err error) {
 
 	return encoded, outErr
 }
+
+// PreparedStatementFieldTracker track and replace DataType for column and param ColumnDefinition
+type PreparedStatementFieldTracker struct {
+	proxyHandler *Handler
+	// shared value that indicates number of param packet
+	paramsCounter int
+	columnsNum    uint16
+}
+
+// NewPreparedStatementFieldTracker create new PreparedStatementFieldTracker
+func NewPreparedStatementFieldTracker(handler *Handler, columnNum uint16) PreparedStatementFieldTracker {
+	return PreparedStatementFieldTracker{
+		proxyHandler: handler,
+		columnsNum:   columnNum,
+	}
+}
+
+// ParamsTrackHandler ResponseHandler used to track prepared statement params
+func (p *PreparedStatementFieldTracker) ParamsTrackHandler(ctx context.Context, packet *Packet, _, clientConnection net.Conn) error {
+	clientSession := base.ClientSessionFromContext(ctx)
+	if clientSession == nil {
+		p.proxyHandler.logger.Warningln("No packet without ClientSession in context")
```

Comment:
hm, so in case of error we will not return anything and will not send columns/fields?